internal/service/ipam/model_ipamsvc_host_address.go

```diff
@@ -53,38 +85,109 @@ func ExpandIpamsvcHostAddress(ctx context.Context, o types.Object, diags *diag.D
 	return m.Expand(ctx, diags)
 }
 
-func (m *IpamsvcHostAddressModel) Expand(ctx context.Context, diags *diag.Diagnostics) *ipam.IpamsvcHostAddress {
+func (m *IpamsvcHostAddressModel) Expand(_ context.Context, _ *diag.Diagnostics) *ipam.IpamsvcHostAddress {
 	if m == nil {
 		return nil
 	}
-	to := &ipam.IpamsvcHostAddress{
-		Address: flex.ExpandString(m.Address),
-		Ref:     flex.ExpandString(m.Ref),
-		Space:   flex.ExpandString(m.Space),
+	to := &ipam.IpamsvcHostAddress{}
+	
+	if !m.NextAvailableId.IsNull() && !m.NextAvailableId.IsUnknown() {
+		naip := flex.ExpandString(m.NextAvailableId) + "/nextavailableip"
+		to.Ref = &naip
+	} else {
+		to.Address = flex.ExpandStringPointer(m.Address)
+		to.Space = flex.ExpandStringPointer(m.Space)
 	}
 	return to
 }
 
-func FlattenIpamsvcHostAddress(ctx context.Context, from *ipam.IpamsvcHostAddress, diags *diag.Diagnostics) types.Object {
-	if from == nil {
-		return types.ObjectNull(IpamsvcHostAddressAttrTypes)
+func FlattenIpamsvcHostAddress(ctx context.Context, from []ipam.IpamsvcHostAddress, diags *diag.Diagnostics, data []IpamsvcHostAddressModel) types.List {
+	if len(from) == 0 {
+		return types.ListNull(types.ObjectType{AttrTypes: IpamsvcHostAddressAttrTypes})
 	}
-	m := IpamsvcHostAddressModel{}
-	m.Flatten(ctx, from, diags)
-	t, d := types.ObjectValueFrom(ctx, IpamsvcHostAddressAttrTypes, m)
+	
+	tfData := ApplyToAll(from, data, func(t ipam.IpamsvcHostAddress, e IpamsvcHostAddressModel) types.Object {
+		m := IpamsvcHostAddressModel{}
+		m.Flatten(ctx, &t, diags)
+		m.NextAvailableId = e.NextAvailableId
+		o, d := types.ObjectValueFrom(ctx, IpamsvcHostAddressAttrTypes, m)
+		diags.Append(d...)
+		return o
+	})
+	
+	tfList, d := types.ListValueFrom(ctx, types.ObjectType{AttrTypes: IpamsvcHostAddressAttrTypes}, tfData)
 	diags.Append(d...)
-	return t
+	
+	return tfList
 }
 
-func (m *IpamsvcHostAddressModel) Flatten(ctx context.Context, from *ipam.IpamsvcHostAddress, diags *diag.Diagnostics) {
+func (m *IpamsvcHostAddressModel) Flatten(_ context.Context, from *ipam.IpamsvcHostAddress, _ *diag.Diagnostics) {
 	if from == nil {
 		return
 	}
+	
 	if m == nil {
 		*m = IpamsvcHostAddressModel{}
 	}
-	m.Address = flex.FlattenString(from.Address)
-	m.Ref = flex.FlattenString(from.Ref)
-	m.Space = flex.FlattenString(from.Space)
+	
+	m.Address = flex.FlattenStringPointer(from.Address)
+	m.Ref = flex.FlattenStringPointer(from.Ref)
+	m.Space = flex.FlattenStringPointer(from.Space)
+	
+}
 
+func (m *IpamsvcHostAddressModel) FromTerraform5Value(val tftypes.Value) error {
```

Comment:
I would keep the flatten function to work on a single object. In the new function you can pass the Model as well, so that we don't create a new model everytime. 

A separate FlattenList function can be created, or existing FlattenList function can be enhanced to support this